### ISSUE TITLE
[BUG] incorrect codegen for operators on struct pointers 

### ISSUE THREAD

**Issue by maxbachmann** [cython/cython#4801]
**Describe the bug**
For the operators `+=`/`-=`/`*=`/`/=`/`%=`/`^=`/`&=`/`|=` invalid code is generated when using them on pointers to `structs`/`classes`. Currently the following code:
```cython
from cython.operator cimport dereference

ctypedef struct Test:
    int a

def test():
    cdef Test* b = NULL
    dereference(b).a += 1
```
generates the following C/C++ code:
```c
__pyx_t_4test_Test __pyx_t_1;
__pyx_t_1 = (*__pyx_v_b);
__pyx_t_1.a = (__pyx_t_1.a + 1);
```
this means the assignment is only performed on the temporary and therefor has no effect.

**Expected behavior**
There are a couple of ways to improve this:
1) these operators are already forbidden for direct usage with dereference:
```cython
def test():
    cdef int* a = NULL;
    dereference(a) += 1;
```
currently leads to the following error `Illegal operand for inplace operation.`. It would be certainly better to prevent the usage than generating broken code.
2) I do not see any reason why this temporary even has to be generated here, so the code gen e.g. for `+=` could be
```c
(*__pyx_v_b).a = ((*__pyx_v_b) + 1);
```
3) the code gen from 2) is strictly speaking still not correct in C++, since the operators `+` and `+=` can have different behaviors. Even though this is rare, this means it would be best to avoid the transformation from `+=` to `+`:
```c
(*__pyx_v_b).a += 1;
```

**Environment (please complete the following information):**
 - OS: any
 - Python version any
 - Cython version: 3.0.0a10

**Comment by da-woods** [comment]
Yes probably is a bug but you shouldn't need the dereference at all:

```
def test():
    cdef Test* b = NULL
    b.a += 1
```

should work I think

**Comment by maxbachmann** [comment]
This will not work since b is a pointer so the c code has to be:
```c
Test* b = NULL;
b->a += 1
```
or
```c
Test* b = NULL;
(*b).a += 1
```
which is easier to generate. Current workaround is:
```cython
dereference(b).a = dereference(b).a  + 1
```

**Comment by da-woods** [comment]
Cython automatically generates arrow operator on pointers. It works for me at least (just tested it)

**Comment by maxbachmann** [comment]
> Cython automatically generates arrow operator on pointers. It works for me at least (just tested it)

Oh I was only thinking about writing it the way I would in C/C++. That allows for some cleanup in my code base :partying_face:
